src/main/java/co/refiere/dao/CampaignDao.java

```diff
@@ -68,6 +68,42 @@ public void deleteCampaign(Campaign campaign) {
             throw re;
         }
     }
+    
+    public List<Campaign> getReferencesCodesNull(int companyId) {
+        log.debug("getting references_codes instances");
+        try {
+            Session session = sessionFactory.getCurrentSession();
+            org.hibernate.Transaction trans= session.beginTransaction();
+            if(trans.getStatus().equals(TransactionStatus.NOT_ACTIVE))
+                log.debug(" >>> Transaction close.");
+            Query query = session.createSQLQuery("select rf.campaign_id, rf.person_id, rf.referrer_id, rf.code from campaign c join references_codes rf on c.id = rf.campaign_id where rf.referrer_id IS NULL and c.company_id =:companyId");
+            query.setParameter("companyId", companyId);
+            java.util.List<Campaign> results = query.list();
+            trans.commit();
+            return results;
+        } catch (RuntimeException re) {
+            log.error("get failed", re);
+            throw re;
+        }
+    }
+    
+    public List<Campaign> getReferencesCodesNotNull(int companyId) {
```

Comment:
Again, change it to use something more descriptive. Change `getReferencesCodesNotNull` to something like: `getRedeemedCodes`